src/diagnostics.rs

```diff
@@ -114,6 +114,9 @@ pub fn report_error<'tcx, 'mir>(
     };
 
     e.print_backtrace();
+    if let UndefinedBehavior(UndefinedBehaviorInfo::InvalidUndefBytes(Some(ptr))) = e.kind {
+        ecx.memory.dump_alloc(ptr.alloc_id);
+    }
```

Comment:
I am not sure about the placement, maybe it would be better to print this below `report_msg`?

Also in terms of structure, now this "paragraph" of code (separated by empty lines) does two totally unrelated things, which is not great.